apps/language_server/lib/language_server/ex_unit_test_tracer.ex

```diff
@@ -96,6 +104,22 @@ defmodule ElixirLS.LanguageServer.ExUnitTestTracer do
               }
             end)
 
+          tests =
+            grouped_tests
+            |> Map.get(:doctest, [])
+            |> case do
+              [doctest | _] ->
+                %{
+                  name: "doctest #{inspect(doctest.tags.doctest)}",
+                  line: doctest.tags.line - 1,
+                  type: :doctest
+                }
+                |> then(&[&1 | tests])
+
+              _ ->
+                tests
+            end
```

Comment:
this can be made much cleaner by removing the pipelines
```suggestion
           case grouped_tests do 
             %{doctest: [doctest | _]} -> 
               test_meta = %{
                  name: "doctest #{inspect(doctest.tags.doctest)}",
                  line: doctest.tags.line - 1,
                  type: :doctest
                }
                [test_meta | tests]
              _ ->
                tests
            end
```